lib/rubocop/cop/style/closing_array_brace_layout.rb

```diff
@@ -0,0 +1,87 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks that the closing brace in an array is symmetrical
+      # with respect to the opening brace and the array elements.
+      #
+      # If an array's opening brace is on the same line as the first element
+      # of the array, then the closing brace should be on the same line as
+      # the last element of the array.
+      #
+      # If an array's opening brace is on a separate line from the first
+      # element of the array, then the closing brace should be on the line
+      # after the last element of the array.
+      #
+      # @example
+      #
+      #     # bad
+      #     [ :a,
+      #       :b
+      #     ]
+      #
+      #     # bad
+      #     [
+      #       :a,
+      #       :b ]
+      #
+      #     # good
+      #     [ :a,
+      #       :b ]
+      #
+      #     #good
+      #     [
+      #       :a,
+      #       :b
+      #     ]
+      class ClosingArrayBraceLayout < Cop
+        SAME_LINE_MESSAGE = 'Closing array brace must be on the same line as ' \
+          'the last array element when opening brace is on the same line as ' \
+          'the first array element.'
+
+        NEW_LINE_MESSAGE = 'Closing array brace must be on the line after ' \
+          'the last array element when opening brace is on a separate line ' \
+          'from the first array element.'
+
+        def on_array(node)
+          return unless node.loc.begin
```

Comment:
Normally I add comments on guard clauses, so casual readers would know why it's good idea to exit at this point.